### PR TITLE
Overloads + smaller changes in `AutoModRuleAction`

### PR DESCRIPTION
**Pull request by Puncher1**
## Summary

This PR has following changes:
* Add overloads to `AutoModRuleAction.__init__`
* Move `self.type` assignments and use type hint
* Removed unused variable in `from_data`
* Add note to documentation, that only one of `channel_id`, `duration`, or `custom_message` can be used

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/automod.py [hunk 1]
```diff
@@ -81,37 +84,48 @@ class AutoModRuleAction:
...
+    @overload
+    def __init__(self, channel_id: int = None) -> None:
+        ...
+
+    @overload
+    def __init__(self, duration: datetime.timedelta = None) -> None:
+        ...
+
+    @overload
+    def __init__(self, custom_message: str = None) -> None:
```
Comment (Gobot1234) [inline discord/automod.py:96]: These shouldn't have defaults here

**Review by Gobot1234 — APPROVED** [review]

File: discord/automod.py [hunk 2]
```diff
@@ -81,33 +84,43 @@ class AutoModRuleAction:
...
         channel_id: Optional[int] = None,
         duration: Optional[datetime.timedelta] = None,
         custom_message: Optional[str] = None,
     ) -> None:
+        self.type: AutoModRuleActionType = AutoModRuleActionType.block_message
+        if channel_id:
+            self.type = AutoModRuleActionType.send_alert_message
+        elif duration:
+            self.type = AutoModRuleActionType.timeout
+
```
Comment (Rapptz) [inline discord/automod.py:111]: What's the point of this refactoring?

Comment (Puncher1) [reply discord/automod.py:111]: Well, actually there's no particular reason.

**Review by Rapptz — APPROVED** [review]
Thanks. I'll do a future commit to bring back the symmetry the old code had.

**Pull request merged**